server-common/src/main/java/org/apache/kafka/server/storage/log/FetchParams.java

```diff
@@ -42,6 +43,18 @@ public FetchParams(short requestVersion,
                        int maxBytes,
                        FetchIsolation isolation,
                        Optional<ClientMetadata> clientMetadata) {
+        this(requestVersion, replicaId, replicaEpoch, maxWaitMs, minBytes, maxBytes, isolation, clientMetadata, false);
+    }
+
+    public FetchParams(short requestVersion,
+                       int replicaId,
+                       long replicaEpoch,
+                       long maxWaitMs,
+                       int minBytes,
+                       int maxBytes,
+                       FetchIsolation isolation,
+                       Optional<ClientMetadata> clientMetadata,
+                       boolean shareFetchRequest) {
```

Comment:
I wonder whether this is a bit literal. Why are we supplying a `ClientMetadata` here in the share fetch case? That seems to me to be concerned with fetch-from-follower. If we didn't supply a `ClientMetadata`, then `fetchOnlyLeader()` would already return true without needing the new flag.

Reply:
So ClientMetadata includes details for rackId, listenerName which I see in regular fetch if utilized for figuring read replica. Not sure if that would be relevant for ShareFetch in future and if we should skip always for ShareFetch. Hence I have added the additional shareFetchBoolean. Please let me know what you think. Also adding @junrao to provide his inputs.